src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -196,6 +197,31 @@ public ResponseEntity<UserDTO> updateUser(@Valid @RequestBody ManagedUserVM mana
         return ResponseEntity.ok().headers(HeaderUtil.createAlert(applicationName, "userManagement.updated", managedUserVM.getLogin())).body(new UserDTO(updatedUser));
     }
 
+    /**
+     * PUT ldap : Updates an existing User based on the info available in the LDAP server.
+     *
+     * @param login of the user to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated user
+     */
+    @PutMapping("users/{login:" + Constants.LOGIN_REGEX + "}/sync-ldap")
+    @PreAuthorize("hasRole('ADMIN')")
+    @Profile("ldap")
+    public ResponseEntity<UserDTO> syncUserViaLdap(@PathVariable String login) {
+        log.debug("REST request to update ldap information User : {}", login);
+
+        var existingUser = userRepository.findOneWithGroupsAndAuthoritiesByLogin(login);
+
+        if (existingUser.isPresent()) {
+            var ldapUser = userService.updateUserFromLdap(existingUser.get().getRegistrationNumber());
```

Comment:
I don't think this is the intended functionality. What happens in the user's matriculation number is null? This would not work.

Instead I suggest to make this much simpler, have a look how JiraAuthenticationProvider:168 does that. I would propose the following code.

```
ldapUserService.ifPresent(service -> service.loadUserDetailsFromLdap(user));
userCreationService.saveUser(user)
```